### PR TITLE
[WIP] Display and manage Git-based packages

### PR DESCRIPTION
**Pull request by kuychaco**
Signed-off-by: Michelle Tilley <binarymuse@github.com>

### CODE REVIEW

**Comment by BinaryMuse** [comment]
Depends on atom/apm#518. Still under construction and some edge cases to fix up, but thought it'd be nice to get this in front of folks and :sparkles: show off :sparkles: the new hotness so far :fire: 

Settings view displays Git-based packages in a new section:

settings_ __users_mtilley_github_atom

The package card shows the installed revision instead of the download count:

settings_ __users_mtilley_github_atom

Updating works as you'd expect, reporting the current and upstream revisions:

settings-view-git-update

File: lib/package-card.coffee [hunk 1]
```diff
@@ -227,9 +236,12 @@ class PackageCard extends View
...
 
   displayInstalledState: ->
-    if @newVersion
+    if @newVersion or @newSha
       @updateButtonGroup.show()
-      @updateButton.text("Update to #{@newVersion}")
+      if @newVersion
+        @updateButton.text("Update to #{@newVersion}")
+      else if @newSha
+        @updateButton.text("Update to ##{@newSha.substr(0, 8)}")
```
Comment (joshaber) [inline lib/package-card.coffee]: Same nit as https://github.com/atom/apm/pull/518#discussion-diff-56237165 about the meaning of the hash prefix.

File: lib/package-detail-view.coffee [hunk 2]
```diff
@@ -220,7 +220,7 @@ class PackageDetailView extends ScrollView
 
   openMarkdownFile: (path) ->
     if atom.packages.isPackageActive('markdown-preview')
-      atom.workspace.open("#{encodeURI("markdown-preview://#{path}")}")
```
Comment (joshaber) [inline lib/package-detail-view.coffee]: Ha, nice catch :+1:

**Comment by joshaber** [comment]
Seems reasonable to me :+1:

**Pull request merged**